test/irmin/test.ml

```diff
@@ -265,6 +266,80 @@ let test_to_string () =
   test "record" my_record_t
     { foo = 2; flag = false; letter = Delta }
     "{\"foo\":2,\"flag\":0,\"letter\":\"Delta\"}";
+
+  ()
+
+(** Test the behaviour of {!Irmin.Type.pp_ty}. *)
+let test_pp_ty () =
+  let test : type a. string -> a T.t -> string -> unit =
+   fun case_name input expected_output ->
+    let assertion =
+      Fmt.strf "Expected output of `pp_ty` for representation of `%s`" case_name
+    in
+    (Fmt.to_to_string T.pp_ty) input
+    |> Alcotest.(check string) assertion expected_output
+  in
+
+  (* Test cases for basic types *)
+  test "unit" T.unit "Prim Unit";
```

Comment:
Is `Prim` really necessary here? Maybe the printer should just omit these.

Reply:
Agreed :slightly_smiling_face: I've included this in my list of suggestions in the PR description.

I think it's good to test the existing behaviour first.

Reply:
Hum yes I jumped directly to the code without reading the description, feel free to ignore my comments ;-)